homeassistant/components/iotawatt/sensor.py

```diff
@@ -206,13 +228,39 @@ def extra_state_attributes(self):
         attrs = {"type": data.getType()}
         if attrs["type"] == "Input":
             attrs["channel"] = data.getChannel()
+        if self._accumulating:
+            attrs[
+                ATTR_LAST_UPDATE
+            ] = self.coordinator.api.getLastUpdateTime().isoformat()
 
         return attrs
 
+    async def async_added_to_hass(self):
+        """Load the last known state value of the entity if the accumulated type."""
+        await super().async_added_to_hass()
+        if self._accumulating:
+            state = await self.async_get_last_state()
+            self._accumulated_value = 0.0
+            if state:
+                try:
+                    self._accumulated_value = float(state.state)
+                    if ATTR_LAST_UPDATE in state.attributes:
+                        self.coordinator.update_last_run(
+                            dt.parse_datetime(state.attributes.get(ATTR_LAST_UPDATE))
```

Comment:
```suggestion
                            dt.parse_datetime(state.attributes[ATTR_LAST_UPDATE])
```

This can't be missing since you are checking it above